Kernel/VM/Space.cpp

```diff
@@ -202,6 +202,14 @@ Vector<Region*> Space::find_regions_containing(const Range& range)
 
     return regions;
 }
+template<typename Callback>
+void Space::for_each_region_intersecting(const Range& range, Callback callback)
+{
+    for (auto* region : find_regions_intersecting(range)) {
```

Comment:
The idea for using a `for_each_...` was to essentially to not implement `find_regions_intersecting` so that instead of putting the regions in a vector you just iterate them, and for those that do intersect you simply call `callback`.

Reply:
...

Reply:
This isnt viable anymore due to needing two loops in the new implementation due to safety reasons